src/Avalonia.FreeDesktop/Notifications/FreeDesktopNotificationManager.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Avalonia.Controls.Notifications;
+using Tmds.DBus;
+
+namespace Avalonia.FreeDesktop.Notifications
+{
+    public class FreeDesktopNotificationManager : INotificationManager, IDisposable
+    {
+        private const string NotificationsService
+            = "org.freedesktop.Notifications";
+
+        // ReSharper disable once InconsistentNaming
+        private const int DEFAULT_NOTIFICATION_EXPIRATION = -1;
+
+        // ReSharper disable once InconsistentNaming
+        private const int FOREVER_NOTIFICATION_EXPIRATION = 0;
+
+        private static readonly ObjectPath NotificationsPath
+            = new ObjectPath("/org/freedesktop/Notifications");
+
+        private readonly Dictionary<uint, INotification> _notifications;
+
+        private readonly IFreeDesktopNotificationsProxy _proxy;
+        private IDisposable _actionWatcher;
+        private IDisposable _closeNotificationWatcher;
+        private volatile bool _isConnected;
+
+        public FreeDesktopNotificationManager()
+        {
+            _notifications = new Dictionary<uint, INotification>();
+            _proxy = Connection.Session.CreateProxy<IFreeDesktopNotificationsProxy>(
+                NotificationsService,
+                NotificationsPath
+            );
+
+            Connect();
+            SetupWatcherTasks();
+        }
+
+        public void Dispose()
+        {
+            _actionWatcher?.Dispose();
+            _closeNotificationWatcher?.Dispose();
+        }
+
+        public async void Show(INotification notification)
```

Comment:
`async void` is problematic for a number of reasons (Exceptions will pretty much crash the application). We should probably introduce a `ShowAsync()` function in `INotificationManager` and throw on the synchronous one. Or we adapt something like `AsyncHelper.RunSync` (https://cpratt.co/async-tips-tricks/) and just block (blocking time should be fairly short I reckon).

Reply:
Same applies for `SetupWatcherTasks()` and `Connect()`. I don't really want to make the entire `X11Platform` initilization routine `async` just for these two functions.